modules/deser-lib/src/index.ts

```diff
@@ -0,0 +1,144 @@
+import { decodeFirstSync, encodeCanonical } from 'cbor';
```

Comment:
I don't think we should put all of this in the index. 

To make this module more extensible it might be better to have 
- index.ts
- cbor.ts

and in `index.ts` you can export as `export * as Cbor from './cbor';`

Feel free to change the names, but by scoping it this way you can add new serialization schemes without having to change the old function names (in case of conflicts)